src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -162,7 +162,11 @@ private void CreateNodeCaches()
             {
                 if (_compilationModuleGroup.ContainsType(type))
                 {
-                    if (type.IsCanonicalSubtype(CanonicalFormKind.Any))
+                    if (type.IsCanonicalDefinitionType(CanonicalFormKind.Any))
+                    {
+                        return new CanonicalDefinitionEETypeNode(this, type);
```

Comment:
Who is asking for a constructed EEType node for a canonical definition type? This really should be an assert, the same way we assert nobody asks for a constructed generic definition. This thing should never be allocated and doesn't have a vtable. Whoever is asking for this is not getting back what they expect.